src/test/java/com/google/neighborgood/MessageServletTest.java

```diff
@@ -0,0 +1,273 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.neighborgood.servlets;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableMap;
+import java.io.*;
+import javax.servlet.http.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit test on the MessageServlet file */
+@RunWith(JUnit4.class)
+public final class MessageServletTest {
+
+  /* Set up the test environment with Datastore and UserService and simulate the situation
+   * that the user is logged in as admin, has a specific email and userID.
+   */
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("leonardzhang@google.com")
+          .setEnvAuthDomain("1234567890")
+          .setEnvAttributes(
+              ImmutableMap.of(
+                  "com.google.appengine.api.users.UserService.user_id_key", "1234567890"));
+
+  private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+  private final PrintStream originalErr = System.err;
+  private UserService userService;
+  private DatastoreService ds;
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    userService = UserServiceFactory.getUserService();
+    ds = DatastoreServiceFactory.getDatastoreService();
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testEnvironmentTest() {
+    // Test the UserService feature
+    assertTrue(userService.isUserAdmin());
```

Comment:
Why do we assert the user is an admin? Are there any differences in how this works between an admin and a regular user?

Reply:
Actually there are no differences between an admin and a regular user. The assertion here actually just makes sure that the test environment has been set up correctly. When I set up the test environment, I made the mock user as the admin, so I wanted to make sure that the setup worked properly here.